### PR TITLE
Remove service url scheme hardcode in cli

### PR DESCRIPTION
**Pull request by gps949**
Signed-off-by: Chenyang Gao <gps949@outlook.com>

There's hardcode scheme(http://) of KubeFATE service url in Kubefate cli, so that it can only support service without TLS (http & ws).
It's neither graceful nor security.
This commit removes the hardcode and add a new option in the config.yaml. (Modified the README accordingly)

### CODE REVIEW

**Comment by JingChen23** [comment]
Hi, Thanks for the contribution 🙌 ! 
I have a couple of comments WRT this PR:

1. You can only checkin to [FederatedAI:develop-1.9.0](https://github.com/FederatedAI/KubeFATE/tree/develop-1.9.0), not master. I helped you to change the base branch but I find that there is a conflict in the readme file, you need to help to resolve it.
2. You need to add test result after changing the url to https. To my knowledge, the TLS/SSL will not comes from nowhere so I think you need to add a certificate to make it work. I think we also need to test that, and record the steps into the README file. Also, please post the test results into the description for this PR.

**Comment by JingChen23** [comment]
I don't think so:

```
→ kubefate version
* kubefate commandLine version=v1.4.4
* kubefate service connection error, Post "https://example.com:32126/v1/user/login": http: server gave HTTP response to HTTPS client
```

I think you need to rebuild the Kubefate CLI tool first, and put it under the /bin/ dir.

Come on, you are trying to use the CLI to call the server with https url, but you change nothing on the KubeFATE server side, how can this work?

File: k8s-deploy/pkg/cli/kube_log_cli.go [hunk 1]
```diff
@@ -206,10 +211,15 @@ func GetModuleLogFollow(uuid, args string) error {
 	if serviceUrl == "" {
 		serviceUrl = "localhost:8080/"
 	}
-	Url := "ws://" + serviceUrl + "/" + apiVersion + r.Path + fmt.Sprintf("/%s/ws?%s", uuid, args)
+	safeconnect := viper.GetString("safeconnect")
+	scheme := "ws://"
+	if safeconnect == "true" {
+		scheme = "wss://"
```
[the remark this answers is not in the document]
Comment (gps949) [reply k8s-deploy/pkg/cli/kube_log_cli.go:217]: > I don't think so:
> 
> ```
> → kubefate version
> * kubefate commandLine version=v1.4.4
> * kubefate service connection error, Post "https://example.com:32126/v1/user/login": http: server gave HTTP response to HTTPS client
> ```
> 
> I think you need to rebuild the Kubefate CLI tool first, and put it under the /bin/ dir.
> 
> Come on, you are trying to use the CLI to call the server with https url, but you change nothing on the KubeFATE server side, how can this work?

We don’t need to make any change for src code of the service side.
For example, when we're using ingress-nginx, we can set the certificate in the yaml of k8s ingress part. It’s just a deployment config method but not src code change. 
We can also set a reverse proxy in front of the kubefate http service, such as nginx or CDN, without change the kubefate service config yaml.

(Just as Java web application deployment, we can set the ssl & cert only in the Tomcat or Weblogic. If we don't need doing the 2-way ssl to verify the client side cert for blocking illegal user, we need to do nothing in the src code. Middle ware or reverse proxy will handle the 1-way ssl)

**Comment by gps949** [comment]
> I don't think so:
> 
> ```
> → kubefate version
> * kubefate commandLine version=v1.4.4
> * kubefate service connection error, Post "https://example.com:32126/v1/user/login": http: server gave HTTP response to HTTPS client
> ```
> 
> I think you need to rebuild the Kubefate CLI tool first, and put it under the /bin/ dir.
> 
> Come on, you are trying to use the CLI to call the server with https url, but you change nothing on the KubeFATE server side, how can this work?

We don’t need to make any change for src code of the service side.
For example, when we're using ingress-nginx, we can set the certificate in the yaml of k8s ingress part. It’s just a deployment config method but not src code change.
We can also set a reverse proxy in front of the kubefate http service, such as nginx or CDN, without change the kubefate service config yaml.

(Just as Java web application deployment, we can set the ssl & cert only in the Tomcat or Weblogic. If we don't need doing the 2-way ssl to verify the client side cert for blocking illegal user, we need to do nothing in the src code. Middle ware or reverse proxy will handle the 1-way ssl)

File: k8s-deploy/pkg/cli/kube_log_cli.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gps949) [reply k8s-deploy/pkg/cli/kube_log_cli.go:217]: For the "kubefate cluster logs -f" verification, I made a piece of screen recoding.

[kubefate cluster logs test](https://drive.google.com/file/d/1Yf7-TkaNhtTF-H_t3IGwrzW3BIPrY2YV/view?usp=sharing)

For privacy, I mask my password & domain. Nothing else edited in the record.
There's error in the logs output, it should be other issues in server side but not related to our discussion

**Comment by JingChen23** [comment]
Thanks for the explanation. It would be great if in the future you can add a brand new README telling how to add the TLS support in the ingress of kubefate, including the easist way to prepare the certificate and private key, and how you changed the kubefate.yaml file.

It is "It would be great if you can do" but there is no hard requirement or commitment. WRT this change only, I think it looks good so I will approve and merge it.

**Review by JingChen23 — APPROVED** [review]

**Comment by gps949** [comment]
> Thanks for the explanation. It would be great if in the future you can add a brand new README telling how to add the TLS support in the ingress of kubefate, including the easist way to prepare the certificate and private key, and how you changed the kubefate.yaml file.
> 
> It is "It would be great if you can do" but there is no hard requirement or commitment. WRT this change only, I think it looks good so I will approve and merge it.

Thank you for your checking and confirm.
I'd like to add interpretation about the TLS configuration into related docs later. Considering I've got a full-time job to fighting with, I can't give a deadline. I wish it will be done before the 1.9 release :)

**Comment by JingChen23** [comment]
That can't be better, thanks!

**Pull request merged**